### PR TITLE
bump k8s dep to v1.25.5

### PR DESCRIPTION
**Pull request by Huang-Wei**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

- bump k8s dep to v1.25.5
- adapt to k/k interface changes
- add scheduler config v1

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by Huang-Wei** [comment]
/assign @denkensk 

@yibozhuang if you have some cycles, feel free to review as well.

**Comment by Huang-Wei** [comment]
/hold

I need to have https://github.com/kubernetes/test-infra/pull/28283 merged first.

File: go.mod [hunk 1]
```diff
@@ -1,35 +1,40 @@
...
-	k8s.io/utils v0.0.0-20220210201930-3a6ce19ff2f9
+	k8s.io/klog/v2 v2.70.1
+	k8s.io/kube-scheduler v0.25.5
+	k8s.io/kubernetes v1.25.5
+	k8s.io/utils v0.0.0-20220728103510-ee6ede2d64ed
 	sigs.k8s.io/controller-runtime v0.12.3
 	sigs.k8s.io/yaml v1.3.0
 )
 
+require (
```
Comment (denkensk) [inline go.mod:33]: Can we merge the three `require` together?

File: pkg/capacityscheduling/capacity_scheduling_test.go [hunk 2]
```diff
@@ -22,6 +22,7 @@ import (
 	"testing"
 
 	gocmp "github.com/google/go-cmp/cmp"
+	"k8s.io/apimachinery/pkg/util/wait"
```
Comment (denkensk) [inline pkg/capacityscheduling/capacity_scheduling_test.go:26]: remove this blank line？

File: pkg/networkaware/networkoverhead/networkoverhead_test.go [hunk 3]
```diff
@@ -1439,15 +1443,15 @@ func BenchmarkNetworkOverheadFilter(b *testing.B) {
 				if err != nil {
 					b.Fatalf("Failed to create Workload %q: %v", p.Name, err)
 				}
-				//b.Logf("Workload %v created  \n", p.Name)
+				// b.Logf("Workload %v created  \n", p.Name)
```
Comment (denkensk) [inline pkg/networkaware/networkoverhead/networkoverhead_test.go:1446]: remove this line directly？

File: test/util/fake.go [hunk 4]
```diff
@@ -35,6 +35,10 @@ type fakeSharedLister struct {
 	havePodsWithRequiredAntiAffinityNodeInfoList []*framework.NodeInfo
 }
 
+func (f *fakeSharedLister) StorageInfos() framework.StorageInfoLister {
```
Comment (denkensk) [inline test/util/fake.go:38]: move this function to L84? stay with other member function. <(*￣▽￣*)/

**Review by denkensk — COMMENTED** [review]
some small comments

**Comment by denkensk** [comment]
/retest 

> /hold
> 
> I need to have [kubernetes/test-infra#28283](https://github.com/kubernetes/test-infra/pull/28283) merged first.

Merged.
/retest

**Comment by Huang-Wei** [comment]
@denkensk thanks for the review. All comments addressed.

btw: I'd like to merge the doc PR (#463) before this one, so still /hold here.

**Comment by Huang-Wei** [comment]
#463 merged. This PR is ready for review.

/unhold

**Comment by Huang-Wei** [comment]
Kindly ping @denkensk

**Review by yibozhuang — APPROVED** [review]
/lgtm

**Comment by denkensk** [comment]
/lgtm
/approve
Thanks. @Huang-Wei

**Pull request merged**